pootle/apps/pootle_store/unit/filters.py

```diff
@@ -32,21 +34,27 @@ def filter(self, unit_filter):
 
 class UnitChecksFilter(BaseUnitFilter):
 
-    def __init__(self, qs, *args, **kwargs):
-        self.qs = qs
+    def __init__(self, qs, unit_filter, *args, **kwargs):
+        super(UnitChecksFilter, self).__init__(qs, unit_filter, *args, **kwargs)
         self.checks = kwargs.get("checks")
         self.category = kwargs.get("category")
 
     def filter_checks(self):
+        if self.unit_filter != 'checks':
```

Comment:
I'm thinking this whole change to keep track of `unit_filter` is unnecessary, as the `filter` method in `BaseUnit` basically makes sure `filter_<unit_filter>` will be called. Therefore in this specific line (which is the only place where `self.unit_filter` is used) it's assured that `unit_filter == 'checks'` and hence this branch is dead code.